### PR TITLE
[ODS-5174] Total count and paging is off when API Client is associated with Edorgs at multiple level

### PR DESCRIPTION
**Pull request by gmcelhanon**

### CODE REVIEW

**Review by bptillman — APPROVED** [review]
Tested and confirmed the count is correct now and did not find any adverse reactions from these changes being implemented otherwise. Only had my one comment that is not a blocker. Just more of a question to clarify, so I'm approving this.

File: Application/EdFi.Ods.Common/Infrastructure/Repositories/GetEntitiesBySpecification.cs [hunk 1]
```diff
@@ -100,17 +100,19 @@ async Task<SpecificationResult> GetPagedAggregateIdsAsync()
...
-                    ? await queryBatch.GetResultAsync<Guid>(resultIndex++, cancellationToken)
-                    : new Guid[0];
+                    ? (await queryBatch.GetResultAsync<object[]>(resultIndex++, cancellationToken))
+                        .Select(r => (Guid) r[0])
+                        .ToArray()
+                    : Array.Empty<Guid>();
                 
                 var totalCount = CountRequested()
-                    ? (await queryBatch.GetResultAsync<long>(resultIndex, cancellationToken)).First()
+                    ? (await queryBatch.GetResultAsync<int>(resultIndex, cancellationToken)).First()
```
[the remark this answers is not in the document]
Comment (gmcelhanon) [reply Application/EdFi.Ods.Common/Infrastructure/Repositories/GetEntitiesBySpecification.cs:115]: While the `Projections` class provides 32-bit and 64-bit versions for the "row count" methods (through `RowCount` and `RowCountInt64`), there is no `Int64` variant provided for `CountDistinct`. Thus, the value that comes back from NHibernate is an `int`. Practically speaking though, I also don't think that any implementations will approach 2.147B items in any _aggregate root_ table, so I'm not overly concerned.

[the remark this answers is not in the document]
Comment (bptillman) [reply Application/EdFi.Ods.Common/Infrastructure/Repositories/GetEntitiesBySpecification.cs:115]: So then would it make sense for the TotalCount prop on the private SpecificationResult class here be changed to int as well?

[the remark this answers is not in the document]
Comment (gmcelhanon) [reply Application/EdFi.Ods.Common/Infrastructure/Repositories/GetEntitiesBySpecification.cs:115]: For consistency, I went ahead and changed the types on the two other places that the value lands before being serialized to the client.

**Pull request merged**